### PR TITLE
bugfix for timelinewriter

### PR DESCRIPTION
**Pull request by NRauschmayr**
### Description of changes:
more details in https://github.com/awslabs/sagemaker-debugger/pull/454

#### Style and formatting:

I have run `pre-commit install` to ensure that auto-formatting happens with every commit.

#### Issue number, if available

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by leleamol — APPROVED** [review]
Approving the change but can we verify that the hourly directories are getting created event if rotation_policy.file_close_interval != 60

File: smdebug/core/tfevent/timeline_file_writer.py [hunk 1]
```diff
@@ -306,13 +306,15 @@ def _should_rotate_now(self, now_in_us):
...
-            return True
+        #        if diff_in_hours != 0:
+        #            return True
 
         if diff_in_seconds > rotation_policy.file_close_interval:
             return True
 
         if file_size > rotation_policy.file_max_size:
+            if now_in_us != self.last_event_end_time_in_us:
+                self.last_event_end_time_in_us = now_in_us
```
Comment (NihalHarish) [inline smdebug/core/tfevent/timeline_file_writer.py:317]: Can we add a test for these changes?

**Review by NihalHarish — CHANGES_REQUESTED** [review]

File: smdebug/core/tfevent/timeline_file_writer.py [hunk 2]
```diff
@@ -306,13 +306,15 @@ def _should_rotate_now(self, now_in_us):
         file_size, diff_in_seconds, diff_in_hours = self._get_rotation_info(now_in_us)
         rotation_policy = self._profiler_config_parser.config.trace_file.rotation_policy
 
-        if diff_in_hours != 0:
-            return True
+        #        if diff_in_hours != 0:
+        #            return True
```
[the remark this answers is not in the document]
Comment (NRauschmayr) [reply smdebug/core/tfevent/timeline_file_writer.py]: i removed it

File: smdebug/core/tfevent/timeline_file_writer.py (hunk 1 above)
Comment (NRauschmayr) [reply smdebug/core/tfevent/timeline_file_writer.py:314]: This line fixes the problem of overwriting the same filename where we basically lost data during detailed profiling. I have tested this with long running training jobs and here is the result that shows that without the fix merged timeline is missing events:
image


With bugfix:
image

The new testcases that will be added to the CI ensure this part is tested.

**Review by NihalHarish — APPROVED** [review]

**Pull request merged**